docs/articles/movie_recommendation_using_vectordb.markdown

```diff
@@ -0,0 +1,568 @@
+
+Each row in the table corresponds to a single movie, with columns storing data such as title, genres, overview, and embeddings. This columnar format offers a convenient way to store and retrieve information for a variety of tasks related to embeddings. 
+
+## Using Genre Spectrum Embeddings to get the relevant recommendations.
+
+And now, after all the groundwork, we've arrived at the final piece of the puzzle. Let's generate some relevant recommendations using `genre_embeddings` and LanceDB VectorDB.
+
+```python
+movie_data_pd = pd.DataFrame(movie_data)
+def get_recommendation(title):
+    result = (
+        movie_table.search(movie_data_pd[movie_data_pd["title"] == title]["embeddings"].values[0]).metric('cosine')
+        .limit(5)
+        .to_pandas()
+    )
+    return result
+
+result = get_recommendation("Toy Story")
+result[['title']]
+```
+
+Well let me break down some things for you.
+ 
+First, we initialize our LanceDB table containing our relevant data. Then, for a given movie title, we check if we have this title in our dataset. If it exists, we perform a cosine similarity search on all other movies, returning the top 5 most relevant movie titles.
+
+And just like that, you've crafted an impressive movie recommendation system that can swiftly provide you with the best selection of movies. This is made possible by leveraging `genre_embeddings` created with the assistance of a neural network and storing them in LanceDB vector database.
+
+Doing all this will provide you with a list of recommended movies.
+
+*Just to clarify, I've employed a rudimentary deep learning model architecture to showcase how we can improve our recommendation system. While the movie recommendations may not yield optimal results, the team at Tubi would be the experts to provide you with the exact model architecture they've utilized for developing this system.*
+
+That being said, Here's an overview of the entire training code..
+
+```python
+import pandas as pd
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from nltk import word_tokenize
+from torch.utils.data import DataLoader, TensorDataset
+from gensim.models.doc2vec import Doc2Vec, TaggedDocument
+from sklearn.preprocessing import MultiLabelBinarizer
+from sklearn.model_selection import train_test_split
+from tqdm import tqdm
+
+import nltk
+nltk.download('punkt')
+
+# Read data from CSV file
+movie_data = pd.read_csv('data/movies_metadata.csv', low_memory=False)
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+
+def preprocess_data(movie_data_chunk):
+    tagged_docs = []
+    valid_indices = []
+    movie_info = []
+
+    # Wrap your loop with tqdm
+    for i, row in tqdm(movie_data_chunk.iterrows(), total=len(movie_data_chunk)):
```

Comment:
iterrows is probably the least efficient way to work through a dataframe:
check the second answer here: https://stackoverflow.com/questions/16476924/how-can-i-iterate-over-rows-in-a-pandas-dataframe

Reply:
Oh thanks @morkapronczay, I am changing the `iterrows` to something more efficient

Reply:
Thanks, I updated with the list comprehensions and it's damn fast!